src/components/DataTable/cells/TableStructuredCell.types.ts

```diff
@@ -0,0 +1,24 @@
+import type {TableCellProps} from './TableCell.types';
+
+export type TableStructuredCellProps = TableCellProps & {
+    /**
+     * Depth level in the tree hierarchy (0 = root level)
+     * Used for indentation calculation
+     */
+    depth: number;
+
+    /**
+     * Whether this row can be expanded (has children)
+     */
+    isExpandable: boolean;
+
+    /**
+     * Whether this row is currently expanded
+     */
+    isExpanded: boolean;
```

Comment:
Should this prop be optional? Because if the item is not expandable there is no reason to provide `isExpanded`